blivet/blivet.py

```diff
@@ -791,7 +791,12 @@ def create_device(self, device):
         is_snapshot = isinstance(device, LVMLogicalVolumeDevice) and device.is_snapshot_lv
 
         if device.format.type and not device.format_immutable and not is_snapshot:
-            self.devicetree.actions.add(ActionCreateFormat(device))
+            try:
+                self.devicetree.actions.add(ActionCreateFormat(device))
+            except DependencyError:  # pylint: disable=broad-except
+                # revert devicetree changes no matter what happened
+                self.reset_device(device)
```

Comment:
I think we talked about the possibility of canceling only the actions you created/scheduled just before in `create_device` instead of reverting all the way back to the original device. Is that possible?

Reply:
Fixed.